### ISSUE TITLE
Get FILE * from filename with fopen

### ISSUE THREAD

**Issue by mfkl** [mfkl/LibVLCSharp-readonly#9]
[libvlcpp](https://code.videolan.org/videolan/libvlcpp/blob/master/vlcpp/Instance.hpp#L278) offers the ability to set up logging to a specific file. 

[Code](https://github.com/mfkl/libvlcsharp/blob/735eff8ba86d43368ed2cf9fb0a92ab831a60727/LibVlcSharp/Manual/Instance.cs#L391) currently is not working. Should be available crossplatform with Mono too.

**Comment by jeremyVignelles** [comment]
should we use the libvlc function that print to a file or recreate the same feature with .net tools and log callbacks? That would be easier to do, but would be a cheat, not 1-1 aligned with libvlc.

If I remember correctly, FILE is an opaque OS-specific structure

**Comment by mfkl** [comment]
Yeah, not sure how to get an opaque FILE * from C# in a crossplatform way...

Maybe we should expose another API (even if different from libvlcpp, if it doesn't make sense/hard to do from a C# perspective, probably the best choice). What do you think?

**Comment by jeremyVignelles** [comment]
We could use FILE* by calling native `fopen` with IntPtr parameters. For a C# reimplementation, we will need to resolve #5 and be careful of the perfs.

Now that I'm thinking about it, using the native `fopen` seems to be the most realistic, both for perfs and for ease to implement.

**Comment by jeremyVignelles** [comment]
But then, your title is misleading. I don't think we can transform a C# `FileStream` into a C `FILE*`, but we can give a file name to `libvlcsharp` so that it creates a `FILE*` and writes to it.

EDIT: Which is quite useless given that this behavior can be achieved with vlc parameters...

**Comment by mfkl** [comment]
So to achieve this on Windows we have to `DllImport` into `msvcrt.dll`. [Info](http://www.pinvoke.net/default.aspx/msvcrt.fopen).

How about with Mono on macOS and Linux though?

**Comment by mfkl** [comment]
Pretty much done in https://github.com/mfkl/libvlcsharp/commit/fd87dff36d5cee6df2bdfc3c0f5daa800238605f

Untested on Linux/OSX.
File is created and initial log is written to it, but tests fails as file handle is kept until test execution finishes (even after UnsetLog/Instance.Dispose()) and sharing is not enabled.

**Comment by mfkl** [comment]
Added `fclose` in `UnsetLog` as per your suggestion.

Need to fix #14 to test this on OSX/Linux and close if test is green.
